src/transaction/log_global.c

```diff
@@ -159,6 +159,19 @@ log_global::initialize_log_prior_receiver ()
 {
   assert (m_prior_recver == nullptr);
   m_prior_recver = std::make_unique<cublog::prior_recver> (prior_info);
+  
+  if (is_passive_transaction_server ())
```

Comment:
I cannot say I like this approach :) .
This introduces the `is_..._server` conditional fork in yet another of the legacy files whereas there is already a fork in `server_type.cpp` file which directly calls this code.

If you do not prefer adding a new function ... then it's a one-liner in `init_server_type`:
```
...
      else if (is_passive_transaction_server ())
	{
	  assert (pts_Gl == nullptr);

	  // passive tran server also needs (a) prior receiver(s) to receive log from page server(s)
	  // the mechanism is that of a pub sub; a PTS subscribes to receive log from one page server (for now)
	  log_Gl.initialize_log_prior_receiver ();
	  log_Gl.get_log_prior_receiver ().start_thread ();

	  pts_Gl = new passive_tran_server ();
	  ts_Gl.reset (pts_Gl);
	}
...
```

Reply:
Oh, you're right. I didn't consider the call path. Thank you.